### PR TITLE
Wpi testing

### PR DESCRIPTION
**Pull request by BrownGenius**
These are the changes we made at WPI.  This code needs to be cleaned up prior to a push to main, but creating a PR to document the changes.

### CODE REVIEW

File: src/main/deploy/pathplanner/MobilityBlueHumanSideToDock.path
Comment (BrownGenius) [inline src/main/deploy/pathplanner/MobilityBlueHumanSideToDock.path:77]: Updated autonomous routines to start from edge of community

File: src/main/deploy/robotconfig/00802F17DEE0.json
Comment (BrownGenius) [inline src/main/deploy/robotconfig/00802F17DEE0.json:54]: Updated based on sysId run @ WPI.  Not final.

File: src/main/java/bhs/devilbotz/Constants.java
Comment (BrownGenius) [inline src/main/java/bhs/devilbotz/Constants.java:86]: Gripper solenoids were reversed on the actual robot

File: src/main/java/bhs/devilbotz/RobotContainer.java
Comment (BrownGenius) [inline src/main/java/bhs/devilbotz/RobotContainer.java:76]: Only create cameraManager on real robot. _This should probably be handed inside of CameraManager_.

File: src/main/java/bhs/devilbotz/RobotContainer.java
Comment (BrownGenius) [inline src/main/java/bhs/devilbotz/RobotContainer.java:78]: The default gripper command was always triggering, causing the gripper open/close command to fail.

File: src/main/java/bhs/devilbotz/RobotContainer.java
Comment (BrownGenius) [inline src/main/java/bhs/devilbotz/RobotContainer.java:109]: @ParkerMeyers made this change. Not sure the difference between toggleOnTrue vs onTrue.

File: src/main/java/bhs/devilbotz/RobotContainer.java
Comment (BrownGenius) [inline src/main/java/bhs/devilbotz/RobotContainer.java:202]: Added automatic selection of red vs blue paths.

File: src/main/java/bhs/devilbotz/commands/arm/ArmDown.java
Comment (BrownGenius) [inline src/main/java/bhs/devilbotz/commands/arm/ArmDown.java:48]: the `atBottom()` condition was causing the gripper arm to behave erratically when the bottom limit switch was broken/loose.

File: src/main/java/bhs/devilbotz/commands/gripper/GripperIdle.java
Comment (BrownGenius) [inline src/main/java/bhs/devilbotz/commands/gripper/GripperIdle.java:46]: @ParkerMeyers made this change.  Not sure why.

File: src/main/java/bhs/devilbotz/subsystems/Arm.java
Comment (BrownGenius) [inline src/main/java/bhs/devilbotz/subsystems/Arm.java:102]: Empirical arm encoder values for cone vs cube scoring.

File: src/main/java/bhs/devilbotz/subsystems/DriveTrain.java
Comment (BrownGenius) [inline src/main/java/bhs/devilbotz/subsystems/DriveTrain.java:581]: navx on competition bot is rotated 180 compared to the practice bot, so the roll value needs to be negated. _This needs to be moved to the robotconfig json files as a flag.....e.g. rollInversion=true/false_

File: src/main/java/bhs/devilbotz/subsystems/DriveTrain.java
Comment (BrownGenius) [inline src/main/java/bhs/devilbotz/subsystems/DriveTrain.java:604]: In simulation, this code hides the actual robot location.  _should disable only when in simulation.....or make it transparent_

File: src/main/java/bhs/devilbotz/subsystems/Gripper.java
Comment (BrownGenius) [inline src/main/java/bhs/devilbotz/subsystems/Gripper.java:47]: Debug code for gripper

File: src/main/java/bhs/devilbotz/utils/ShuffleboardManager.java
Comment (BrownGenius) [inline src/main/java/bhs/devilbotz/utils/ShuffleboardManager.java:84]: Removed unimplemented routines.  Shortened naming.

File: src/main/java/bhs/devilbotz/subsystems/Gripper.java
Comment (loafdog) [reply src/main/java/bhs/devilbotz/subsystems/Gripper.java:47]: I don't understand what `kOff` does for a solenoid.  We should figure that out and see if the stop method is needed.

File: src/main/java/bhs/devilbotz/RobotContainer.java
Comment (BrownGenius) [reply src/main/java/bhs/devilbotz/RobotContainer.java:78]: We can experiment with cleaning this up Monday when we have the actual robot.  The code as it is seems to do the correct thing.

File: src/main/java/bhs/devilbotz/RobotContainer.java
Comment (BrownGenius) [reply src/main/java/bhs/devilbotz/RobotContainer.java:102]: Yes, the gripper controls were not doing anything.  When the electrical team put a meter on the solenoid leads, they saw a short blip when the button was pressed, so something was prematurely turning the gripper solenoids, off.

File: src/main/java/bhs/devilbotz/commands/arm/ArmToTop.java
Comment (BrownGenius) [reply src/main/java/bhs/devilbotz/commands/arm/ArmToTop.java:46]: Monday, we should pre-program/test the automatic arm height buttons.

File: src/main/java/bhs/devilbotz/commands/gripper/GripperIdle.java
Comment (BrownGenius) [reply src/main/java/bhs/devilbotz/commands/gripper/GripperIdle.java:46]: We can test other options on Monday.

File: src/main/java/bhs/devilbotz/subsystems/Gripper.java
Comment (BrownGenius) [reply src/main/java/bhs/devilbotz/subsystems/Gripper.java:47]: I believe stop turns off both solenoids, so keeps the gripper wherever it is currently. Neither the forward or reverse is actuated.

Comment (loafdog) [reply src/main/java/bhs/devilbotz/subsystems/Gripper.java:47]: Got it.  I think these prints should be removed.  They may fill up console with lots of msgs, or get lost if other print statements fill up console.  We should figure out how to put equivalent status to Shuffleboard.

**Review by loafdog — COMMENTED** [review]
One change I'd like to see, rest lgtm.

File: src/main/java/bhs/devilbotz/subsystems/Gripper.java
Comment (BrownGenius) [reply src/main/java/bhs/devilbotz/subsystems/Gripper.java]: Done. I removed the prints.

Comment (ParkerMeyers) [reply src/main/java/bhs/devilbotz/subsystems/Gripper.java]: yep they were just for quick debugging

File: src/main/java/bhs/devilbotz/Robot.java
Comment (ParkerMeyers) [inline src/main/java/bhs/devilbotz/Robot.java:237]: Javadoc please

File: src/main/java/bhs/devilbotz/Robot.java
Comment (ParkerMeyers) [inline src/main/java/bhs/devilbotz/Robot.java:233]: Javadocs, same with all other methods in Robot.json

File: src/main/java/bhs/devilbotz/RobotContainer.java
Comment (ParkerMeyers) [inline src/main/java/bhs/devilbotz/RobotContainer.java:96]: I know other bindings dont have it, but we should add comments to each binding in order to make it easier to understand the code

File: src/main/java/bhs/devilbotz/subsystems/DriveTrain.java
Comment (ParkerMeyers) [inline src/main/java/bhs/devilbotz/subsystems/DriveTrain.java:291]: Needed?

File: src/main/java/bhs/devilbotz/subsystems/DriveTrain.java
Comment (ParkerMeyers) [inline src/main/java/bhs/devilbotz/subsystems/DriveTrain.java:468]: Do we know why the sensor phase needs to be changed?

**Review by ParkerMeyers — COMMENTED** [review]
Minor changes/Documentation.

Make sure everything has javadocs

**Comment by loafdog** [comment]
Suggested changes to add gripper to shuffleboard here: https://github.com/BHSRobotix/ChargedUp2023/tree/gripper_shuffleboard
I created the gripper_shuffleboard branch off of this one. I did not create a PR because not sure how/when to review it.  We can:
- create PR against this branch, review, and end up merging to this branch, and then merge this branch to main
- wait on creating PR/review until after this branch is merged to main
- ignore the branch if we do not want the changes to shuffleboard at all.

**Review by loafdog — APPROVED** [review]

**Comment by BrownGenius** [comment]
@loafdog I'm going to add the javadocs that Parker requested. I can pull in your branch to this and review it all together.

File: src/main/java/bhs/devilbotz/Robot.java
Comment (BrownGenius) [reply src/main/java/bhs/devilbotz/Robot.java:243]: done

File: src/main/java/bhs/devilbotz/Robot.java
Comment (BrownGenius) [reply src/main/java/bhs/devilbotz/Robot.java:233]: already had javadocs

File: src/main/java/bhs/devilbotz/RobotContainer.java
Comment (BrownGenius) [reply src/main/java/bhs/devilbotz/RobotContainer.java:96]: I agree, we should add comments as part of a separate PR.

File: src/main/java/bhs/devilbotz/subsystems/DriveTrain.java
Comment (BrownGenius) [reply src/main/java/bhs/devilbotz/subsystems/DriveTrain.java:467]: My understanding is the kitbot drivetrain doesn't have a fixed "front" vs "back", so the encoder phase depends on how they are mounted and what we decide "forward" on the robot is.  According to the docs, the phase needs to be determined empirically by driving the robot forward and set the phase such that the both left and right encoder values are positive.

**Review by loafdog — APPROVED** [review]
Thanks for doing all this work,  and for merging in shuffleboard changes.

**Pull request merged**
